src/compiler-documenting.md

```diff
@@ -0,0 +1,49 @@
+# Documenting rustc
+
+You might want to build documentation of the various components 
+available like the standard library. There’s two ways to go about this.
+ You can run rustdoc directly on the file to make sure the HTML is 
+ correct which is fast or you can build the documentation as part of the 
+ build process through x.py. Both are viable methods since documentation 
+ is more about the content.
+
+## Document everything
+
+   `./x.py doc`
+
+## If you want to avoid the whole Stage 2 build
+
+   `./x.py doc --stage 1`
+
+First the compiler and rustdoc get built to make sure everything is okay 
+and then it documents the files.
+
+## Document specific components
+
+```bash
+   ./x.py doc src/doc/book
+   ./x.py doc src/doc/nomicon
+   ./x.py doc src/doc/book src/libstd
+```
+
+Much like individual tests or building certain components you can build only
+ the documentation you want.
+
+## Document internal rustc items
+By default rustc does not build the compiler docs for its internal items. 
```

Comment:
@davidtwco Is this still true? How is the [rustc rustdoc](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/) built?

Reply:
Compiler documentation is not built by default - there's a `config.toml` flag for that. But, when enabled, compiler documentation does [include internal items](https://github.com/rust-lang/rust/blob/79fcc58b24d85743d025fd880fca55748662ed3e/src/bootstrap/doc.rs#L699).